src/main/java/ca/corefacility/bioinformatics/irida/web/controller/api/samples/RESTSampleSequenceFilesController.java

```diff
@@ -422,45 +422,56 @@ public ModelMap addNewSequenceFileToSample(@PathVariable Long sampleId, @Request
-		sequenceFile.add(selfRel);
-		sequenceFile.add(linkTo(methodOn(RESTProjectSamplesController.class).getSample(sampleId)).withRel(REL_SAMPLE));
-		sequenceFile.add(linkTo(
-				methodOn(RESTSampleSequenceFilesController.class).readSequencingObject(sampleId, objectType,
-						singleEndSequenceFile.getId())).withRel(REL_SEQ_OBJECT));
-		
-		modelMap.addAttribute(RESTGenericController.RESOURCE_NAME, sequenceFile);
-		// add a location header.
-		response.addHeader(HttpHeaders.LOCATION, selfRel.getHref());
-		// set the response status.
-		response.setStatus(HttpStatus.CREATED.value());
+		try {
+			// save the seqobject and sample
+			SampleSequencingObjectJoin createSequencingObjectInSample = sequencingObjectService.createSequencingObjectInSample(
+					singleEndSequenceFile, sample);
+
+			singleEndSequenceFile = (SingleEndSequenceFile) createSequencingObjectInSample.getObject();
+			logger.trace("Created seqfile in sample " + createSequencingObjectInSample.getObject()
+					.getId());
+
+			// clean up the temporary files.
+			Files.deleteIfExists(target);
+			Files.deleteIfExists(temp);
+			logger.trace("Deleted temp file");
+			// prepare a link to the sequence file itself (on the sequence file
+			// controller)
+			String objectType = objectLabels.get(SingleEndSequenceFile.class);
+			Long sequenceFileId = singleEndSequenceFile.getSequenceFile()
+					.getId();
+			Link selfRel = linkTo(
+					methodOn(RESTSampleSequenceFilesController.class).readSequenceFileForSequencingObject(sampleId,
+							objectType, singleEndSequenceFile.getId(), sequenceFileId)).withSelfRel();
+
+			// Changed, because sfr.setResource(sf)
+			// and sfr.setResource(sampleSequenceFileRelationship.getObject())
+			// both will not pass a GET-POST comparison integration test.
+			singleEndSequenceFile = (SingleEndSequenceFile) sequencingObjectService.read(singleEndSequenceFile.getId());
+			SequenceFile sequenceFile = singleEndSequenceFile.getFileWithId(sequenceFileId);
+
+			// add links to the resource
+			sequenceFile.add(
+					linkTo(methodOn(RESTSampleSequenceFilesController.class).getSampleSequenceFiles(sampleId)).withRel(
+							REL_SAMPLE_SEQUENCE_FILES));
+			sequenceFile.add(selfRel);
+			sequenceFile.add(
+					linkTo(methodOn(RESTProjectSamplesController.class).getSample(sampleId)).withRel(REL_SAMPLE));
+			sequenceFile.add(
+					linkTo(methodOn(RESTSampleSequenceFilesController.class).readSequencingObject(sampleId, objectType,
+							singleEndSequenceFile.getId())).withRel(REL_SEQ_OBJECT));
+
+			modelMap.addAttribute(RESTGenericController.RESOURCE_NAME, sequenceFile);
+			// add a location header.
+			response.addHeader(HttpHeaders.LOCATION, selfRel.getHref());
+			// set the response status.
+			response.setStatus(HttpStatus.CREATED.value());
+
+		} catch (IllegalArgumentException e) {
+			response.setStatus(HttpStatus.BAD_REQUEST.value());
+		} catch (Exception e) {
+			response.setStatus(HttpStatus.INTERNAL_SERVER_ERROR.value());
```

Comment:
Same comments here about catching `Exception` and logging.

Reply:
Fixed in 1d9c20b27fa35533f4334f521558c3e28baef6ae